static/src/javascripts/projects/commercial/modules/sticky-top-banner.js

```diff
@@ -0,0 +1,210 @@
+                stickyBanner.style.height = `${newHeight}px`;
+                header.style.marginTop = `${newHeight}px`;
+            }
+            if (topSlotHeight !== undefined && headerHeight <= scrollY) {
+                window.scrollBy(0, newHeight - topSlotHeight);
+            }
+            topSlotHeight = newHeight;
+
+            return newHeight;
+        });
+    }
+    return Promise.resolve(-1);
+};
+
+// Sudden changes in the layout can be jarring to the user, so we animate
+// them for a better experience. We only do this if the slot is in view
+// though.
+const setupAnimation = (): Promise<any> =>
+    fastdom.write(() => {
+        if (stickyBanner && header) {
+            if (scrollY <= headerHeight) {
+                header.classList.add('l-header--animate');
+                stickyBanner.classList.add('sticky-top-banner-ad--animate');
+            } else {
+                header.classList.remove('l-header--animate');
+                stickyBanner.classList.remove('sticky-top-banner-ad--animate');
+            }
+        }
+    });
+
+const onScroll = (): Promise<any> => {
+    scrollY = window.pageYOffset;
+    if (!updateQueued) {
+        updateQueued = true;
+
+        return fastdom
+            .write(() => {
+                updateQueued = false;
+                if (stickyBanner) {
+                    if (headerHeight < scrollY) {
+                        stickyBanner.style.position = 'absolute';
+                        stickyBanner.style.top = `${headerHeight}px`;
+                    } else {
+                        stickyBanner.style.position = '';
+                        stickyBanner.style.top = '';
+                    }
+                }
+            })
+            .then(setupAnimation);
+    }
+
+    return Promise.resolve();
+};
+
+const update = (newHeight: number): Promise<any> =>
+    fastdom
+        .read(() => {
+            topSlotStyles = topSlotStyles || window.getComputedStyle(topSlot);
+            return (
+                newHeight +
+                parseInt(topSlotStyles.paddingTop, 10) +
+                parseInt(topSlotStyles.paddingBottom, 10)
+            );
+        })
+        .then(resizeStickyBanner);
+
+const onResize = (specs: any, _: any, iframe: ?Element): void => {
+    if (topSlot && topSlot.contains(iframe)) {
+        update(specs.height);
+        unregister('resize', onResize);
+    }
+};
+
+// Register a message listener for when the creative wants to resize
+// its container
+// We also listen for scroll events if we need to, to snap the slot in
+// place when it reaches the end of the header.
+const setupListeners = (): void => {
+    register('resize', onResize);
+    if (!config.page.hasSuperStickyBanner) {
+        addEventListener(window, 'scroll', onScroll, {
+            passive: true,
+        });
+    }
+};
+
+const getAdvertSizeByIndex = (advert: ?Advert, index: number): ?number => {
+    if (advert && advert.size && typeof advert.size !== 'string') {
```

Comment:
Does the `advert.size` type check add value in this case (asking because [the definition of .size is always string](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/commercial/types.js#L5))?

Could be a case for a post conversion review comment ...